### PR TITLE
Add verification procedure for multicast

### PR DESCRIPTION
**Pull request by jboxman**
Preview link: https://multicast-verification--ocpdocs.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/enabling-multicast.html#nw-enabling-multicast_ovn-kubernetes-enabling-multicast

Related: https://access.redhat.com/solutions/3304441

### CODE REVIEW

**Comment by jboxman** [comment]
Hi @weliang1,

I decided to add a procedure for verifying that multicast works, much like the SCTP verification I worked on earlier in the year. Is this something that you can verify?

Thanks!

**Comment by weliang1** [comment]
> Hi @weliang1,
> 
> I decided to add a procedure for verifying that multicast works, much like the SCTP verification I worked on earlier in the year. Is this something that you can verify?
> 
> Thanks!

Hi @jboxman,

Multicast still not supported in OVN due to:
https://bugzilla.redhat.com/show_bug.cgi?id=1843695

**Comment by jboxman** [comment]
@weliang1, how about for OpenShift SDN?

- https://multicast-verification--ocpdocs.netlify.app/openshift-enterprise/latest/networking/openshift_sdn/enabling-multicast.html

**Comment by weliang1** [comment]
> @weliang1, how about for OpenShift SDN?
> 
> * https://multicast-verification--ocpdocs.netlify.app/openshift-enterprise/latest/networking/openshift_sdn/enabling-multicast.html

@jboxman, I need test the multicast server and client YAML file from your configuration sample, will update results soon, thanks!

**Comment by weliang1** [comment]
@jboxman  Can not install multicast receiver pod by using YAML file from your doc
`[weliang@weliang verification-tests]$ cat <<EOF| oc create -f -
> apiVersion: v1
> kind: Pod
> metadata:
>   name: mlistener
>   labels:
>     app: multicast-verify
> spec:
>   containers:
>     - name: mlistener
>       image: registry.access.redhat.com/ubi8
>       command: ["/bin/sh", "-c"]
>       args:
>         ["dnf install socat hostname && sleep inf"]
>       ports:
>         - containerPort: 30102
>           name: mlistener
>           protocol: UDP
> EOF
pod/mlistener created
[weliang@weliang verification-tests]$ oc get pods
NAME        READY   STATUS              RESTARTS   AGE
mlistener   0/1     ContainerCreating   0          3s
[weliang@weliang verification-tests]$ oc get pods
NAME        READY   STATUS              RESTARTS   AGE
mlistener   0/1     ContainerCreating   0          6s
[weliang@weliang verification-tests]$ oc get pods
NAME        READY   STATUS   RESTARTS   AGE
mlistener   0/1     Error    0          11s
[weliang@weliang verification-tests]$ oc describe pod mlistener
Name:         mlistener
Namespace:    test
Priority:     0
Node:         ip-10-0-168-76.us-east-2.compute.internal/10.0.168.76
Start Time:   Thu, 23 Jul 2020 11:39:27 -0400
Labels:       app=multicast-verify
Annotations:  k8s.v1.cni.cncf.io/network-status:
                [{
                    "name": "openshift-sdn",
                    "interface": "eth0",
                    "ips": [
                        "10.128.2.5"
                    ],
                    "default": true,
                    "dns": {}
                }]
              k8s.v1.cni.cncf.io/networks-status:
                [{
                    "name": "openshift-sdn",
                    "interface": "eth0",
                    "ips": [
                        "10.128.2.5"
                    ],
                    "default": true,
                    "dns": {}
                }]
              openshift.io/scc: anyuid
Status:       Running
IP:           10.128.2.5
IPs:
  IP:  10.128.2.5
Containers:
  mlistener:
    Container ID:  cri-o://86cb544a7270b9f3e02acf3e7a7e1d408771557bbe3dd3ef8d62178c884bc604
    Image:         registry.access.redhat.com/ubi8
    Image ID:      registry.access.redhat.com/ubi8@sha256:74493b300ffa7098a5d54da0611925b69bd6899c0860774fe0565bf9ec0976fe
    Port:          30102/UDP
    Host Port:     0/UDP
    Command:
      /bin/sh
      -c
    Args:
      dnf install socat hostname && sleep inf
    State:          Running
      Started:      Thu, 23 Jul 2020 11:39:55 -0400
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Thu, 23 Jul 2020 11:39:38 -0400
      Finished:     Thu, 23 Jul 2020 11:39:40 -0400
    Ready:          True
    Restart Count:  2
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-9hqjl (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  default-token-9hqjl:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-9hqjl
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason          Age               From                                                Message
  ----     ------          ----              ----                                                -------
  Normal   Scheduled       <unknown>         default-scheduler                                   Successfully assigned test/mlistener to ip-10-0-168-76.us-east-2.compute.internal
  Normal   AddedInterface  26s               multus                                              Add eth0 [10.128.2.5/23]
  Warning  BackOff         14s               kubelet, ip-10-0-168-76.us-east-2.compute.internal  Back-off restarting failed container
  Normal   Pulling         1s (x3 over 26s)  kubelet, ip-10-0-168-76.us-east-2.compute.internal  Pulling image "registry.access.redhat.com/ubi8"
  Normal   Pulled          1s (x3 over 21s)  kubelet, ip-10-0-168-76.us-east-2.compute.internal  Successfully pulled image "registry.access.redhat.com/ubi8"
  Normal   Created         0s (x3 over 21s)  kubelet, ip-10-0-168-76.us-east-2.compute.internal  Created container mlistener
  Normal   Started         0s (x3 over 21s)  kubelet, ip-10-0-168-76.us-east-2.compute.internal  Started container mlistener
[weliang@weliang verification-tests]$ oc get pods
NAME        READY   STATUS   RESTARTS   AGE
mlistener   0/1     Error    2          33s
[weliang@weliang verification-tests]$ oc logs mlistener
Updating Subscription Management repositories.
Unable to read consumer identity
Subscription Manager is operating in container mode.
This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
Red Hat Universal Base Image 8 (RPMs) - BaseOS  5.4 MB/s | 767 kB     00:00    
Red Hat Universal Base Image 8 (RPMs) - AppStre  22 MB/s | 3.9 MB     00:00    
Red Hat Universal Base Image 8 (RPMs) - CodeRea 113 kB/s |  11 kB     00:00    
Dependencies resolved.
================================================================================
 Package        Architecture Version                Repository             Size
================================================================================
Installing:
 hostname       x86_64       3.20-6.el8             ubi-8-baseos           32 k
 socat          x86_64       1.7.3.3-2.el8          ubi-8-appstream       302 k

Transaction Summary
================================================================================
Install  2 Packages

Total download size: 334 k
Installed size: 1.2 M
Is this ok [y/N]: Operation aborted.
[weliang@weliang verification-tests]$ oc get pod
NAME        READY   STATUS             RESTARTS   AGE
mlistener   0/1     CrashLoopBackOff   2          53s
[weliang@weliang verification-tests]$ 
`

**Comment by weliang1** [comment]
@jboxman Can not install multicast receiver pod by using YAML file from your doc.

The testing log: https://privatebin-it-iso.int.open.paas.redhat.com/?4885b4ae2cbc8eb5#eA1A534/uxsAOfxY20JDxUGljUtplQnVJBBSc9Wbnu0=

**Comment by jboxman** [comment]
@weliang1, oh, looks like it needs `dnf -y install socat hostname && sleep inf` I think, because it prompts:

```
Total download size: 334 k
Installed size: 1.2 M
Is this ok [y/N]: Operation aborted.
```

File: modules/nw-enabling-multicast.adoc [hunk 1]
```diff
@@ -24,15 +24,111 @@ You can enable multicast between Pods for your project.
...
+  name: mlistener
+  labels:
+    app: multicast-verify
+spec:
+  containers:
+    - name: mlistener
+      image: registry.access.redhat.com/ubi8
+      command: ["/bin/sh", "-c"]
+      args:
+        ["dnf install socat hostname && sleep inf"]
```
Comment (weliang1) [inline modules/nw-enabling-multicast.adoc:63]: Correct command is: ["dnf -f install socat hostname && sleep inf"]

File: modules/nw-enabling-multicast.adoc [hunk 2]
```diff
@@ -24,15 +24,111 @@ You can enable multicast between Pods for your project.
...
+  name: msender
+  labels:
+    app: multicast-verify
+spec:
+  containers:
+    - name: msender
+      image: registry.access.redhat.com/ubi8
+      command: ["/bin/sh", "-c"]
+      args:
+        ["dnf install socat && sleep inf"]
```
Comment (weliang1) [inline modules/nw-enabling-multicast.adoc:88]: Correct command is: ["dnf -f install socat hostname && sleep inf"]

File: modules/nw-enabling-multicast.adoc [hunk 3]
```diff
@@ -24,15 +24,111 @@ You can enable multicast between Pods for your project.
...
+EOF
+----
+
+. Start the multicast listener:
+
+.. Get the IP address for the Pod:
++
+[source,terminal]
+----
+$ POD_IP=$(oc get pods mserver -o jsonpath='{.status.podIP}')
```
Comment (weliang1) [inline modules/nw-enabling-multicast.adoc:98]: Correct command is: POD_IP=$(oc get pods mlistener -o jsonpath='{.status.podIP}')

File: modules/nw-enabling-multicast.adoc [hunk 4]
```diff
@@ -24,15 +24,111 @@ You can enable multicast between Pods for your project.
...
+----
+$ CIDR=$(oc get Network.config.openshift.io cluster \
+    -o jsonpath='{.status.clusterNetwork[0].cidr}')
+----
+
+.. To send a multicast message, enter the following command:
++
+[source,terminal]
+----
+$ oc exec mclient -i -t -- \
```
Comment (weliang1) [inline modules/nw-enabling-multicast.adoc:123]: Need update to use: oc exec msender -i -t --     /bin/bash -c "echo | socat STDIO UDP4-DATAGRAM:224.1.0.1:30102,range=$CIDR"

**Review by weliang1 — COMMENTED** [review]
> @weliang1, oh, looks like it needs `dnf -y install socat hostname && sleep inf` I think, because it prompts:
> 
> ```
> Total download size: 334 k
> Installed size: 1.2 M
> Is this ok [y/N]: Operation aborted.
> ```

Correct, the YAML file need update to include -y

File: modules/nw-enabling-multicast.adoc [hunk 5]
```diff
@@ -24,15 +24,111 @@ You can enable multicast between Pods for your project.
...
+  containers:
+    - name: msender
+      image: registry.access.redhat.com/ubi8
+      command: ["/bin/sh", "-c"]
+      args:
+        ["dnf -y install socat && sleep inf"]
+EOF
+----
+
+. Start the multicast listener:
```
Comment (adellape) [inline modules/nw-enabling-multicast.adoc:92]: Step 4's lead-in sentence here ends with a colon before the substeps, while step 5's lead-in ends with a period. Suggest syncing up on one or the other (personally I use the latter).

**Review by adellape — APPROVED** [review]
One small nit but LGTM!

**Comment by jboxman** [comment]
@weliang1, thanks!

**Pull request merged**